lib/improver/cli/nowcast_extrapolate.py

```diff
@@ -146,20 +163,47 @@ def main(argv=None):
             raise ValueError("Require exactly one output file name for each "
                              "forecast lead time")
 
+    # determine whether accumulations are also to be returned.
+    time_interval = args.lead_time_interval
+    if args.accumulation_fidelity > 0:
+        time_interval = np.gcd(args.lead_time_interval,
+                               args.accumulation_fidelity)
```

Comment:
This statement is choosing the largest number which can be used to construct both the `lead_time_interval` (output rate requirement) and the `accumulation_fidelity` (resolution of accumulations to sum)?  Which will often, but not always, be the minimum of the two numbers (ie we're expecting `accumulation_fidelity=1` and `lead_time_interval=15` as a rule).

Reply:
Correct. Find the largest common factor and use that to reduce the number of advection steps to the minimum possible to achieve the desired outputs.

Reply:
No removed anyway.